browser/ui/views/omnibox/brave_omnibox_view_views.cc

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/omnibox/brave_omnibox_view_views.h"
+
+#include "base/feature_list.h"
+#include "brave/browser/ui/toolbar/brave_location_bar_model_delegate.h"
+#include "brave/common/url_constants.h"
+#include "components/omnibox/browser/omnibox_edit_model.h"
+#include "content/public/common/url_constants.h"
+#include "ui/base/clipboard/scoped_clipboard_writer.h"
+#include "url/gurl.h"
+
+namespace {
+// Copied from omnibox_view_views.cc
+constexpr base::Feature kOmniboxCanCopyHyperlinksToClipboard{
+    "OmniboxCanCopyHyperlinksToClipboard", base::FEATURE_ENABLED_BY_DEFAULT};
+
+bool ShouldAdjust(const base::string16& original,
+                  const base::string16& adjusted) {
+  // Only adjust scheme is changed from brave to chrome.
+  if (GURL(original).scheme() == kBraveUIScheme &&
+      GURL(adjusted).scheme() == content::kChromeUIScheme) {
+    return true;
+  }
+
+  return false;
+}
+
+void ReplaceChromeSchemeToBrave(GURL* url) {
+  DCHECK_EQ(content::kChromeUIScheme, url->scheme());
+  GURL::Replacements replacements;
+  replacements.SetSchemeStr(kBraveUIScheme);
+  *url = url->ReplaceComponents(replacements);
+}
+
+}  // namespace
+
+void BraveOmniboxViewViews::OnAfterCutOrCopy(
+    ui::ClipboardBuffer clipboard_buffer) {
+  ui::Clipboard* cb = ui::Clipboard::GetForCurrentThread();
+  base::string16 selected_text;
+  cb->ReadText(clipboard_buffer, &selected_text);
+  const base::string16 original_selected_text = selected_text;
+
+  GURL url;
+  bool write_url = false;
+  model()->AdjustTextForCopy(GetSelectedRange().GetMin(), &selected_text, &url,
```

Comment:
I'm ok with the adjust back to `brave` scheme approach but seems like `AdjustTextForCopy` change the clipboard text url scheme from `brave` to `chrome`, is it possible to prevent that from happening?

Reply:
I think it's difficult because `AdjustTextForCopy()` just use urls from `controller()->GetLocationBarModel()->GetURL()`.
It was hard to modify the return value of `GetURL()`.

Or do you mean adding replace logic into `AdjustTextForCopy()` like below?
I also considered to modify `AdjustTextForCopy()` but I chose to subclass `OmniboxViewViews` to avoid patching.
```
  if (!user_input_in_progress_ &&
      (*text == display_text_ || *text == url_for_editing_)) {
    *url_from_text = controller()->GetLocationBarModel()->GetURL();
    // Replace chrome  scheme with brave scheme?
    *write_url = true;

    // If the omnibox is displaying a URL, set the hyperlink text to the URL's
    // spec. This undoes any URL elisions.
    if (!controller()->GetLocationBarModel()->GetDisplaySearchTerms(nullptr))
      *text = base::UTF8ToUTF16(url_from_text->spec());

    return;
  }
```

Reply:
Actually I meant `GetURL()`. If that is the case, let's keep current solution

Reply:
Yup, we've tried to return brave scheme url as a VirtualURL(). but it introduced many regressions.